src/helpers/taskHelper.js

```diff
@@ -112,9 +112,15 @@ const taskHelper = function () {
           );
 
           sharedTeamsResult.forEach((_myTeam) => {
+            let hasTeamVisibility = false;
             _myTeam.members.forEach((teamMember) => {
-              if (!teamMember.userId.equals(userid)) teamMemberIds.push(teamMember.userId);
+              if (teamMember.userId.equals(userid) && teamMember.visible) hasTeamVisibility = true;
```

Comment:
As long as the user that is requesting is the last user and was set as visible in settings, they could see all other users as well as their tasks. You didn't check if the user has necessary permission (i.e. if they're an Owner/Administrator/Manager).